cpp/ql/test/query-tests/Likely Bugs/Memory Management/AllocaInLoop/AllocaInLoop1.cpp

```diff
@@ -0,0 +1,84 @@
+// semmle-extractor-options: --clang
+struct vtype {
+	int i1, i2;
+};
+extern int w1, w2;
+
+void *__builtin_alloca(int sz);
+#define alloca __builtin_alloca
+typedef unsigned long size_t;
+
+int printf(const char *format, ...);
+void *memcpy(void *dst, const void *src, size_t len);
```

Comment:
My second comment from https://github.com/Semmle/ql/pull/1094#discussion_r265041635 still applies.

Reply:
C++ does not support 'restrict', though there is GCC's and Clang's __restrict__ , and Visual C++'s __declspec(restrict).  I'll define 'restrict' via macros and update this PR.

Reply:
Yeah, as I said in the comment we often remove `restrict`, but the types and parameter names should match so that the source is clear.

Reply:
@geoffw0 , so is your complaint that the parameters are named 'dst', 'src' and 'len' instead of 's1', 's2' and 'n'?  I've added in the 'restrict', the diff shown above is outdated.

Reply:
Yep.  When it comes to these kinds of rules it's the text of the code that matters, not the meaning of the code.

Reply:
Can you point me to the ISO document(s) you're using here?  I'll happily change the names to what you specified, but I'd like to know what the reference document is moving forward.

FWIW, cplusplus.com lists the function as
<pre>void * memcpy ( void * destination, const void * source, size_t num );</pre>
whereas die.nat has
<pre>void *memcpy(void *dest, const void *src, size_t n);</pre>

Reply:
Rename committed.

Reply:
> Can you point me to the ISO document(s) you're using here?

I've added them to the team drive in the folder `C++/Language Docs`, along with some other docs I sometimes use for reference.  The one you need right now is 'ISO_C_n1124.pdf'.  You can also take headers from the 'Cpp2011' one when you need C++ stuff.